lib/dualis/ui/exam_results_page/exam_results_page.dart

```diff
@@ -87,66 +88,94 @@ class ExamResultsPage extends StatelessWidget {
           alignment: Alignment.topCenter,
         );
       },
-      child: DataTable(
+      child: Column(
         key: ValueKey("semester_${viewModel.currentSemester?.name}"),
-        columnSpacing: 10,
-        dataRowHeight: 60,
-        headingRowHeight: 50,
-        rows: buildGradesDataRows(context, viewModel),
-        columns: buildDataTableColumns(context),
+        children: buildModulesDataTables(context, viewModel),
       ),
       duration: Duration(milliseconds: 200),
     );
   }
 
-  List<DataRow> buildGradesDataRows(
+  List<DataTable> buildModulesDataTables(
       BuildContext context, StudyGradesViewModel viewModel) {
-    var dataRows = <DataRow>[];
+    var dataTables = <DataTable>[];
 
+    var isFirstModule = true;
     for (var module in viewModel.currentSemester.modules) {
-      for (var exam in module.exams) {
-        dataRows.add(
-          DataRow(
-            cells: <DataCell>[
-              DataCell(Column(
-                mainAxisAlignment: MainAxisAlignment.center,
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: <Widget>[
-                  Text(
-                    module.name ?? "",
-                  ),
-                  Text(
-                    exam.name ?? "",
-                    style: Theme.of(context).textTheme.caption,
-                  ),
-                  Text(
-                    exam.semester ?? "",
-                    style: Theme.of(context).textTheme.caption,
-                  ),
-                ],
-              )),
-              DataCell(Text(module.credits)),
-              DataCell(Text(exam.grade.toString())),
-            ],
-          ),
-        );
-      }
+      dataTables.add(DataTable(
+        columnSpacing: 10,
+        dataRowHeight: 60,
```

Comment:
I would set the `dataRowHeight` to 45.
Then more exam results fit onto one screen and the user does not have to scroll too much.